.github/workflows/reusable-solidity-docs.yml

```diff
@@ -0,0 +1,248 @@
+          transforming the Solidity files to an HTML file. The command will be
+          executed in the path specified by `projectSubfolder`."
+        type: string
+        required: false
+      addTOC:
+        description: "True if you want to add Table Of Contents to the generated
+          Markdown file. Uses `markdown-toc` tool to generate TOC. Shouldn't be
+          set to `true` if Docgen's `pages` attribute in `hardhat.config.ts` is
+          set to `items` or `files`."
+        type: boolean
+        required: false
+        default: true
+      tocOptions:
+        description: "Options to be passed to the `markdown-toc` tool used for
+          generation of Table Of Contents. See
+          https://github.com/jonschlinkert/markdown-toc#cli for more details."
+        type: string
+        required: false
+        default: "-i --maxdepth 2"
+      publish:
+        description: "True if you want to push the generated file(s) to the
+          destination repository."
+        type: boolean
+        required: false
+        default: true
+      verifyCommits:
+        description: "True if you want to sign commits updating the docs using
+          GPG key. You'll need to specify XXX and YYY secrets."
+        type: boolean
+        required: false
+        default: false
+      destinationRepo:
+        description: "A name of the repository where the generated Solidity docs
+          will be pushed to. Required if `publish=true`."
+        type: string
+        required: false
+      destinationFolder:
+        description: "A path in the destination repository where the generated
+          Solidity docs will be pushed to. For example, `./docs-api/tbtc-v2`."
+        type: string
+        required: false
+        default: "."
+      destinationBranch:
+        description: "A branch on the destination repository where the generated
+          Solidity docs will be pushed to (must exist)."
+        type: string
+        required: false
+        default: "main"
+      userEmail:
+        description: "The email address of a GitHub user associated with the
+          `githubToken` secret. Will be used to sign commits. Required if
+          `publish=true`."
+        type: string
+        required: false
+      userName:
+        description: "The name of the GitHub user associated with the
+          `githubToken` secret. Will be used to sign commits. Required if
+          `publish=true`."
+        type: string
+        required: false
+      rsyncDelete:
+        description: "True if you want to delete from the destination folder
+          the files which were not generated during current run of the workflow.
+          Usually there shouldn't be need to set this option to `true` if
+          Docgen's `pages` attribute in `hardhat.config.ts` is set to `single`.
+          In other cases the option should be use with caution."
+        type: boolean
+        required: false
+        default: false
```

Comment:
I'm a little bit worried about this option 😨

Reply:
Yeah, I know. That's why I wrote it needs to be used with caution :P
I introduced it to help with handling a situation where we configure Docgen to generate the docs in a way that each contract (or each it's item) produces a separate `.md` file. If such config is used, it would be nice to have a mechanism that removes the files documenting the contracts/items that have been removed from the codebase.
Without this option, the rsync will only add new files / update existing, it would not touch the files that are already in target directory, but don't exist in the source.

But so far the projects which I configured to use the action in all use a setting of Docgen where all the docs land in one common `.md` file. When this config is used, there's no need to worry about redundant files.
So what I think we can do now is to comment out the support for `rsyncDelete` option. And if we'll see it's needed in the future, we can quickly bring it back.